Pinta/MainWindow.cs

```diff
@@ -393,17 +393,56 @@ private void MainWindow_DeleteEvent (object o, DeleteEventArgs args)
 
 		private void MainWindow_DragDataReceived (object o, DragDataReceivedArgs args)
 		{
+			// TODO: Generate random name for the picture being downloaded
+
 			// Only handle URIs
 			if (args.Info != 100)
 				return;
 
 			string fullData = System.Text.Encoding.UTF8.GetString (args.SelectionData.Data);
 
-			foreach (string individualFile in fullData.Split ('\n')) {
-				string file = individualFile.Trim ();
-
-				if (file.StartsWith ("file://"))
-					PintaCore.Workspace.OpenFile (new Uri (file).LocalPath);
+			try {
+				PintaCore.Workspace.HasOpenPendingDocuments = true;
+
+				foreach (string individualFile in fullData.Split ('\n')) {
+					string file = individualFile.Trim ();
+					Console.WriteLine (file);
+
+					if (file.StartsWith ("http") || file.StartsWith("ftp")) {
+						System.Net.WebClient client = new System.Net.WebClient ();
+						string tempFilePath = System.IO.Path.GetTempPath () + System.IO.Path.GetFileName(file);
+
+						var progressDialog = PintaCore.Chrome.ProgressDialog;
+
+						try {
+							progressDialog.Show ();
+							progressDialog.Title = "Downloading file";
+							progressDialog.Text = "File " + file + " is being downloaded to " 
+								+ tempFilePath;
+							
+							client.DownloadFile (file, @tempFilePath);
+							client.DownloadProgressChanged += (sender, e) => 
```

Comment:
I think the issue is the `DownloadFile` blocks until the download is complete. You probably want to look into using `DownloadFileAsync` or `DownloadFileTaskAsync`